lib/helpers.js

```diff
@@ -127,13 +132,56 @@ const parseResults = (editor, results, filePath, showIgnored) => {
   return toReturn;
 };
 
+function requireCached(request) {
+  const modulePath = require.resolve(request);
+  if (moduleCache.has(modulePath)) {
+    return moduleCache.get(modulePath);
+  }
+  // eslint-disable-next-line import/no-dynamic-require
+  const moduleInstance = require(modulePath);
+  moduleCache.set(modulePath, moduleInstance);
+  return moduleInstance;
+}
+
+export function getStylelintPath(filePath) {
+  try {
+    return resolve('stylelint', { basedir: path.dirname(filePath) });
+  } catch (e) {
+    return stylelintLocalPath;
+  }
+}
+
+export function refreshModulesPath(modulesDir) {
+  if (lastModulesPath !== modulesDir) {
+    lastModulesPath = modulesDir;
+    process.env.NODE_PATH = modulesDir || '';
+    // eslint-disable-next-line no-underscore-dangle
+    require('module').Module._initPaths();
+  }
+}
+
+export function getStylelintInstance(filePath) {
+  const stylelintPath = getStylelintPath(filePath);
```

Comment:
The caching should be handled here, not in `requireCached`. As this code is currently written it will _always_ search out the path to `stylelint` which is an expensive operation.

The change to the project path will get things a bit more manageable as there will be fewer unique paths coming into this function, but we should still be caching here to reduce the work done as much as possible.